File: Lab 2/Ethan/topic.py
from kafka.admin import KafkaAdminClient, NewTopic

try:
    admin = KafkaAdminClient(
        bootstrap_servers="localhost:9093", 
        client_id='myapp'
    )

    topics = []
    topics.append(NewTopic(name='Demo1', num_partitions=3, replication_factor=1))
    admin.create_topics(new_topics=topics, validate_only=False)
    print("Topic Created")
    print("Topics:", admin.list_topics())
except:
    print("Soemthing went wrong")    

File: Lab 2/Ethan/topic.js
const Kafka =  require("kafkajs").Kafka

run();
async function run(){
    try{
        const kafka = new Kafka({
            "clientId": "myapp",
            "brokers": ["localhost:9093", "localhost:9094","localhost:9095"]
        })

        const admin = kafka.admin();
        console.log("Connecting....")
        await admin.connect();
        console.log("Connected!")
        await admin.createTopics({
            "topics": [{
                "topic" : "Users",
                "numPartitions": 2
            }]
        })
        console.log("Done!")
        await admin.disconnect();
    }

    catch(ex)
    {
        console.error(`Something bad happened ${ex}`)
    }
}